akka-docs/rst/scala/code/docs/http/scaladsl/HttpServerExampleSpec.scala

```diff
@@ -380,4 +381,36 @@ class HttpServerExampleSpec extends WordSpec with Matchers {
       }
     }
   }
+
+  "all-positive-numbers" in compileOnlySpec {
+    //#all-positive-numbers
+    import akka.stream.scaladsl._
+    import akka.util.ByteString
+    import akka.http.scaladsl.Http
+    import akka.http.scaladsl.model.HttpEntity
+    import akka.http.scaladsl.server.Directives._
+    import akka.stream.ActorMaterializer
+
+    implicit val system = ActorSystem()
+    implicit val materializer = ActorMaterializer()
+
+    // streams are re-usable so we can define it here and use it for every request
+    val numbers = Source.fromIterator(() => Iterator.iterate(1)(n => n + 1))
+
+    val route =
+      path("numbers") {
+        get {
+          complete {
+            HttpEntity(
+              ContentTypes.`text/plain(UTF-8)`,
+              // transform each number to a chunk of bytes
+              numbers.map(n => ByteString(s"$n\n"))
```

Comment:
Not sure I agree on the hello world, the example is the third in the docs with the first of them being a "hello world" example.

The ask an actor/request per actor is a good idea!